### PR TITLE
Add middleware for no cache header and obfuscate server name

### PR DESCRIPTION
**Pull request by richtier**
image

### CODE REVIEW

File: core/middleware.py [hunk 1]
```diff
@@ -60,3 +60,20 @@ def __call__(self, request):
...
+    """
+
+    HEADER_VALUE = "no-store, no-cache, must-revalidate"
+
+    def __init__(self, get_response=None):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        response["Cache-Control"] = self.HEADER_VALUE
```
Comment (r4vi) [inline core/middleware.py:78]: wouldn't it be better to use django's built in `add_never_cache_headers(response)`?

File: core/middleware.py [hunk 2]
```diff
@@ -60,3 +60,20 @@ def __call__(self, request):
...
         return self.get_response(request)
+
+
+class NoCacheMiddlware:
+    """Tell the browser to not cache the pages, because otherwise information that should be kept private can be
+    viewed by anyone with access to the files in the browser's cache directory.
+
+    """
+
+    HEADER_VALUE = "no-store, no-cache, must-revalidate"
```
Comment (alixedi) [inline core/middleware.py:71]: Might consider using (IIRC) [add_never_cache_headers](https://docs.djangoproject.com/en/3.1/ref/utils/#django.utils.cache.add_never_cache_headers), which comes with Django.

Comment (alixedi) [reply core/middleware.py:71]: Something along [these](https://stackoverflow.com/a/5882033/1182202) lines.

File: core/middleware.py (hunk 1 above)
Comment (alixedi) [reply core/middleware.py:78]: inb4! well played 😂

**Review by r4vi — APPROVED** [review]

**Pull request merged**